### ISSUE TITLE
Very slow cache restoration

### ISSUE THREAD

**Issue by iNikem** [actions/cache#381]
Restoring cache sometimes works quite fast and sometimes extremely slow.

E.g. fast restoration:
https://github.com/iNikem/opentelemetry-java-instrumentation/runs/889712406?check_suite_focus=true
<img width="652" alt="Screenshot 2020-07-20 at 15 43 44" src="https://user-images.githubusercontent.com/3010154/87938916-ec392400-ca9f-11ea-8379-3991f96a4a64.png">

Very slow restoration of the same cache
https://github.com/iNikem/opentelemetry-java-instrumentation/runs/889790407?check_suite_focus=true
<img width="663" alt="Screenshot 2020-07-20 at 15 43 28" src="https://user-images.githubusercontent.com/3010154/87938952-f9561300-ca9f-11ea-9772-5f89096125ab.png">

**Comment by timbze** [comment]
See pinned issue #267

**Comment by zw963** [comment]
Same issue.

https://github.community/t/github-action-cache-is-really-slow/123624

https://aws1.discourse-cdn.com/github/original/2X/4/424e19d52919b4460f521c1dd911c3cb30fb82c6.png
![image](https://user-images.githubusercontent.com/549126/88007940-a03ebb80-cb41-11ea-8e00-9798bf9b60ad.png)

But occasionally, it works as expected.

![image](https://user-images.githubusercontent.com/549126/88008006-cfedc380-cb41-11ea-8adc-6ad03d9f4326.png)

**Comment by meiswjn** [comment]
Same issue here. 27 secs vs. 13 minutes in consecutive runs.

**Comment by dhadka** [comment]
Hi all, this has been an ongoing issue.  Closing this in favor of https://github.com/actions/cache/issues/267.

Yesterday we released a new version `@v2.1.0` (the `@v2` tag also points to this release) that switches from using the Actions `http-client` to the Azure Storage SDK for NodeJS.  Unlike the `http-client` which downloads the file as one large stream, the Storage SDK splits the download into 4 MB chunks.  We put in aggressive timeouts (30 sec) for each chunk and allow retries, which will hopefully fix any slow connections.

For example, looking at a recent run from @iNikem's repo, if you're using the new version you'll see:

```

Received 0 of 1682754204 (0.0%), 0.0 MBs/sec
Received 92274688 of 1682754204 (5.5%), 44.0 MBs/sec
Received 218103808 of 1682754204 (13.0%), 69.3 MBs/sec
Received 335544320 of 1682754204 (19.9%), 80.0 MBs/sec
Received 461373440 of 1682754204 (27.4%), 88.0 MBs/sec
Received 583008256 of 1682754204 (34.6%), 92.6 MBs/sec
Received 675282944 of 1682754204 (40.1%), 91.9 MBs/sec
Received 775946240 of 1682754204 (46.1%), 92.4 MBs/sec
Received 868220928 of 1682754204 (51.6%), 91.9 MBs/sec
Received 964689920 of 1682754204 (57.3%), 91.9 MBs/sec
Received 1052770304 of 1682754204 (62.6%), 91.2 MBs/sec
Received 1124073472 of 1682754204 (66.8%), 89.2 MBs/sec
Received 1195376640 of 1682754204 (71.0%), 87.6 MBs/sec
Received 1262485504 of 1682754204 (75.0%), 85.9 MBs/sec
Received 1342177280 of 1682754204 (79.8%), 85.3 MBs/sec
Received 1413480448 of 1682754204 (84.0%), 84.2 MBs/sec
Received 1480589312 of 1682754204 (88.0%), 83.0 MBs/sec
Received 1547698176 of 1682754204 (92.0%), 81.9 MBs/sec
Received 1619001344 of 1682754204 (96.2%), 81.2 MBs/sec
Received 1678559900 of 1682754204 (99.8%), 80.0 MBs/sec
Received 1678559900 of 1682754204 (99.8%), 76.2 MBs/sec
Received 1682754204 of 1682754204 (100.0%), 51.1 MBs/sec
Cache Size: ~1605 MB (1682754204 B)
```

If you continue to see slowness, please post in https://github.com/actions/cache/issues/267.
